### PR TITLE
Added LOW_END define, added Armv8 A35 (PSC) platform and improved OOTB config for slow devices

### PR DESCRIPTION
**Pull request by BananaRama203**
- Added Classic ARMv8A35 platform
- Added DLOW_END to make and assigned it to rPi, A7A7 and A8A35. This sets the config correctly for slower devices so games are actually somewhat playable OOTB.
- I set the default RES to 320x240 for LOW_END (this includes the RPI!) Mainly because this allows for games to run much better without much impact. Only that text is slightly skewed but still readable. 

IMO this is a perfectly acceptable trade off. (More pixelated text vs more smoother performance.)

Maybe can be broken into a separate define called DLOW_RES for rpi however the PSC is more powerful than the rPi3b and I will default it to low res anyway for PSC.

still needs a frame skip option before release for 3d intense games like Crazy Taxi and RECV however with LOW_END and frameskip 1 then the games should be some what playable.

### CODE REVIEW

**Comment by inactive123** [comment]
@flyinghead Leaving it up to you to review this and decide on whether or not this should be merged.

**Comment by 5schatten** [comment]
@swingflip 
Can your add AML S905 based systems as platform too?

File: core/libretro/libretro.cpp [hunk 1]
```diff
@@ -353,11 +365,19 @@ void retro_set_environment(retro_environment_t cb)
...
+#endif
       },
       {
          "reicast_broadcast",
-         "Broadcast; PAL_M|PAL_N|NTSC|PAL|Default",
+#ifdef LOW_END
+         "Broadcast; Default|PAL_M|PAL_N|NTSC|PAL",
+#else
+         "Broadcast; PAL_M|PAL_N|NTSC|PAL|Default",	
+#endif
```
Comment (flyinghead) [inline core/libretro/libretro.cpp]: I think `Default` should be the default setting for all platforms. I'm not sure why `PAL_M` is currently used instead.

File: core/libretro/libretro.cpp [hunk 2]
```diff
@@ -353,11 +365,19 @@ void retro_set_environment(retro_environment_t cb)
       },
       {
          "reicast_cable_type",
+#ifdef LOW_END
+         "Cable type; VGA (RGB)|TV (RGB)|TV (Composite)",	
+#else
          "Cable type; TV (RGB)|TV (Composite)|VGA (RGB)",
+#endif
```
Comment (flyinghead) [inline core/libretro/libretro.cpp:372]: Does this really impact low-end platforms?
I'm not against changing the default setting (Some games don't like `TV (RGB)`) but I'm trying to limit the conditionals

File: core/libretro/libretro.cpp [hunk 3]
```diff
@@ -416,7 +444,11 @@ void retro_set_environment(retro_environment_t cb)
 #if !defined(TARGET_NO_THREADS)
       {
          "reicast_threaded_rendering",
-         "Threaded rendering (restart); disabled|enabled",
+#ifdef LOW_END
+         "Threaded rendering (restart); enabled|disabled",
+#else
+				 "Threaded rendering (restart); disabled|enabled",
+#endif
```
Comment (flyinghead) [inline core/libretro/libretro.cpp]: `Threaded rendering` should be the default for all platforms.

**Review by flyinghead — CHANGES_REQUESTED** [review]

File: core/libretro/libretro.cpp (hunk 1 above)
Comment (BananaRama203) [reply core/libretro/libretro.cpp]: I will do this global.

File: core/libretro/libretro.cpp (hunk 2 above)
Comment (BananaRama203) [reply core/libretro/libretro.cpp:372]: I seemed to have better luck with it but for classic platforms I think this is ok to change.

File: core/libretro/libretro.cpp (hunk 3 above)
Comment (BananaRama203) [reply core/libretro/libretro.cpp]: like above I will fix for all

**Comment by BananaRama203** [comment]
@flyinghead I am also going to break the LOW_RES into it's own definition and remove it from the RPi as default. Although it will make it run much better, I don't think people already running it would appreciate it OOTB however this is a different case for Classic users as they don't know any better. 

If you feel otherwise then by all means just define LOW_RES in the rpi platform too.

**Review by flyinghead — APPROVED** [review]

**Pull request merged**